submissions/vakulinina/js-oop/app.js

```diff
@@ -0,0 +1,117 @@
+const fieldWidth = 505;
+const fieldTopPadding = 50;
+const stepX = 101;
+const stepY = 83;
+const playerStartY = 4 * stepY + fieldTopPadding;
+const playerStartX = 2 * stepX;
+const enemySize = 80;
+const enemyStartX = -stepX;
+const enemyEndX = fieldWidth + stepX;
+const enemyMinSpeed = 100;
+const enemyMaxSpeed = 400;
+const numberOfEnemies = 5;
+let keysEnabled = true;
+let score = 0;
+
+const scoreBoard = document.createElement('h2');
+scoreBoard.style = 'margin: 30px 0 0 0; font-family: Arial, sans-serif;'
+scoreBoard.textContent = `Your score: ${score}`;
+document.body.append(scoreBoard);
+
+const Enemy = function (y, speed) {
+    this.x = enemyStartX;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function (dt) {
+    this.x += this.speed * dt;
+    if (this.x > enemyEndX) {
+        this.x = enemyStartX;
+    }
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Player = function (x, y) {
+    this.x = x;
+    this.y = y;
+    this.sprite = 'images/char-princess-girl.png';
+};
+
+Player.prototype.update = function () {
+    this.checkCollisions();
+};
+
+Player.prototype.checkCollisions = function () {
+    allEnemies.forEach(enemy => {
```

Comment:
Classes should not not refer to any global variables, like global variable `player`, which is an instance of `Player` class (referring to global constants and globals provided by the gaming platform like `Resources` is OK)

Reply:
I'm not sure I completely understand what I should do here. Make `player` a constant?

Reply:
@vakulinina 
`player` was referred to here just for an example. In this particular case "illegal" global variable is `allEnemies`.
Do you see a way to make it so that `Player` class dosn't refer to it directly?
In other words, how we can make an instance of `Player` learn about `allEnemies`?
Hint: how `Player` instance learns about its initial X and Y positions and wouldn't we use `playerStartX` for example directly in `Player` class' methods?

Reply:
Check out the changes, please.
If it's correct, then I have one more question. 
If I assigned a global variable to a property in the constructor directly without passing it as an argument, would it be 'illegal' as well?
Like so: 
`const Player = function (x, y) {
    this.x = x;
    this.y = y;
    this.enemies = allEnemies;
};`

Reply:
Yes, it gonna be illegal as well.
The legal way is just the same Player learns about its X and Y positions.

Reply:
Thank you a lot. You did not approve the PR, is there anything else?